easybuild/tools/module_naming_scheme/toolchain_mns.py

```diff
@@ -0,0 +1,90 @@
+##
+# Copyright 2013-2014 Ghent University
+#
+# This file is part of EasyBuild,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://vscentrum.be/nl/en),
+# the Hercules foundation (http://www.herculesstichting.be/in_English)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# http://github.com/hpcugent/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+Implementation of an example hierarchical module naming scheme.
+
+@author: Kenneth Hoste (Ghent University)
+@author: Markus Geimer (Forschungszentrum Juelich GmbH)
+"""
+
+import os
+import re
+from vsc.utils import fancylogger
+
+from easybuild.tools.module_naming_scheme.hierarchical_mns import HierarchicalMNS
+from easybuild.tools.toolchain import DUMMY_TOOLCHAIN_NAME
+
+CORE = 'Core'
+TOOLCHAIN = 'Toolchain'
+
+MODULECLASS_TC = 'toolchain'
+
+
+class ToolchainMNS(HierarchicalMNS):
+    """Class implementing a toolchain-based hierarchical module naming scheme."""
+
+    def det_module_subdir(self, ec):
+        """
+        Determine module subdirectory, relative to the top of the module path.
+        This determines the separation between module names exposed to users, and what's part of the $MODULEPATH.
+        Examples: Core, Toolchain/gpsolf/2015.02
+        """
+
+        if ec.toolchain.name == DUMMY_TOOLCHAIN_NAME:
+
+            # toolchain is dummy/dummy, put in Core
+            subdir = CORE
+        else:
+            subdir = os.path.join(TOOLCHAIN,ec.toolchain.name,ec.toolchain.version)
+
```

Comment:
get rid of the empty lines in between here, they don't help in making things more readable